### PR TITLE
Add lcov and generate coverage report

### PR DESCRIPTION
**Pull request by anpaskin**
Add lcov and generate coverage report

### CODE REVIEW

File: .codespaces/run-cascade-tests.sh [hunk 1]
```diff
@@ -1,4 +1,7 @@
 #!/bin/bash
 
 dotnet test ~/workspace/vscs-in-codespaces/cascade /p:CodeCoverage=true
-find ~/workspace/vscs-in-codespaces/cascade -name *-lcov.info -exec echo -a {} \; | xargs lcov -o ~/workspace/vscs-in-codespaces/Workspaces/merged-lcov.info
\ No newline at end of file
+find ../cascade -name *-lcov.info -exec echo -a {} \; | xargs lcov -o ../TestCoverage/merged-lcov.info
+reportgenerator -reports:../**/*-lcov.info -targetdir:../TestCoverage -reporttypes:Html
+code ../TestCoverage/index.html
```
Comment (rajkumar42) [inline .codespaces/run-cascade-tests.sh:6]: Does this work? Does it open raw html or rendered html?

Comment (rajkumar42) [reply .codespaces/run-cascade-tests.sh:6]: sorry i didn't see the next line :)

File: .codespaces/run-cascade-tests.sh [hunk 2]
```diff
@@ -1,4 +1,7 @@
 #!/bin/bash
 
 dotnet test ~/workspace/vscs-in-codespaces/cascade /p:CodeCoverage=true
-find ~/workspace/vscs-in-codespaces/cascade -name *-lcov.info -exec echo -a {} \; | xargs lcov -o ~/workspace/vscs-in-codespaces/Workspaces/merged-lcov.info
\ No newline at end of file
+find ../cascade -name *-lcov.info -exec echo -a {} \; | xargs lcov -o ../TestCoverage/merged-lcov.info
+reportgenerator -reports:../**/*-lcov.info -targetdir:../TestCoverage -reporttypes:Html
```
Comment (rajkumar42) [inline .codespaces/run-cascade-tests.sh:5]: reportgenerator -reports:../cascade/**/*-lcov.info -targetdir:../TestCoverage -reporttypes:Html

?

File: .codespaces/run-vsclk-core-tests.sh [hunk 3]
```diff
@@ -1,4 +1,7 @@
 #!/bin/bash
 
 dotnet test ~/workspace/vscs-in-codespaces/vsclk-core /p:CodeCoverage=true
-find ~/workspace/vscs-in-codespaces/vsclk-core -name *-lcov.info -exec echo -a {} \; | xargs lcov -o ~/workspace/vscs-in-codespaces/Workspaces/merged-lcov.info
\ No newline at end of file
+find ../vsclk-core -name *-lcov.info -exec echo -a {} \; | xargs lcov -o ../TestCoverage/merged-lcov.info
+reportgenerator -reports:../**/*-lcov.info -targetdir:../TestCoverage -reporttypes:Html
```
Comment (rajkumar42) [inline .codespaces/run-vsclk-core-tests.sh:5]: same? narrow on vsclk-core

File: .codespaces/run-cascade-tests.sh (hunk 2 above)
Comment (anpaskin) [reply .codespaces/run-cascade-tests.sh:5]: Ah good catch, fixing...

Comment (rajkumar42) [reply .codespaces/run-cascade-tests.sh:5]: Hmm.. I see, you want to merge all coverages into one?  Fine by me. 

Ignore my previous comment.

File: .codespaces/run-vsclk-core-tests.sh (hunk 3 above)
Comment (rajkumar42) [reply .codespaces/run-vsclk-core-tests.sh:5]: Ignore above.

File: .codespaces/run-cascade-tests.sh (hunk 2 above)
Comment (anpaskin) [reply .codespaces/run-cascade-tests.sh:5]: Well if you run the run-cascade-tests task you probably just want to see coverage from the Cascade tests (not from previous runs of vsclk-core tests), so we should scope to Cascade here.

**Review by rajkumar42 — APPROVED** [review]

**Review by rajkumar42 — APPROVED** [review]

**Pull request merged**